rbd/rbd_options.go

```diff
@@ -0,0 +1,122 @@
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// #include <stdlib.h>
+// #include <rbd/librbd.h>
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+)
+
+const (
+	// RBD image options.
+	RbdImageOptionFormat            = C.RBD_IMAGE_OPTION_FORMAT
+	RbdImageOptionFeatures          = C.RBD_IMAGE_OPTION_FEATURES
+	RbdImageOptionOrder             = C.RBD_IMAGE_OPTION_ORDER
+	RbdImageOptionStripeUnit        = C.RBD_IMAGE_OPTION_STRIPE_UNIT
+	RbdImageOptionStripeCount       = C.RBD_IMAGE_OPTION_STRIPE_COUNT
+	RbdImageOptionJournalOrder      = C.RBD_IMAGE_OPTION_JOURNAL_ORDER
+	RbdImageOptionJournalSplayWidth = C.RBD_IMAGE_OPTION_JOURNAL_SPLAY_WIDTH
+	RbdImageOptionJournalPool       = C.RBD_IMAGE_OPTION_JOURNAL_POOL
+	RbdImageOptionFeaturesSet       = C.RBD_IMAGE_OPTION_FEATURES_SET
+	RbdImageOptionFeaturesClear     = C.RBD_IMAGE_OPTION_FEATURES_CLEAR
+	RbdImageOptionDataPool          = C.RBD_IMAGE_OPTION_DATA_POOL
+	// introduced with Ceph Mimic
+	//RbdImageOptionFlatten = C.RBD_IMAGE_OPTION_FLATTEN
```

Comment:
I have some thoughts on handling new apis/ deprecated apis that I mentioned in [a comment on issue 42](https://github.com/ceph/go-ceph/issues/42#issuecomment-548119560).
If we aren't going to use new APIs right away we can probably just leave this as a comment for this PR, but if you want this PR to include this new constant we need to sort out some of the compatibility stuff IMO.

Reply:
Keeping it commented out for now is fine, I could not get a nice alternative to work well.

This needs to be addressed though. An other PR will add support for watchers of an RBD image, and those functions in `librdb` are only available from Ceph Mimic. I do not yet know how to dynamically ignore some `.go` files when building.

Reply:
OK, I will try to make my ideas concrete in an RFC PR. Using your changes as a base would be nice as it can more easily demonstrate my thinking... I will try to jump on that soonish